src/repository/soldierRepo.ts

```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import logger from '../logger';
+import { getDatabase } from './mongoConnect';
+
+interface Soldier {
+  _id?:any;
+  id: string;
+  name: string;
+  rank: string;
+  limitations:Array<string>;
+  duties?:Array<string>;
+}
+
+export async function createSoldier(soldier: Soldier) {
+  try {
+    // eslint-disable-next-line no-underscore-dangle, no-param-reassign
+    soldier._id = +soldier.id;
+    await getDatabase().collection('soldiers').insertOne(soldier);
```

Comment:
Save the collection in a variable - it'll be more readable.

in addition the connection to the collection takes time (I think), so it would save time...

Reply:
I couldn't manage to make this work